### PR TITLE
Add dictionary for attribution scope fields.

### PR DESCRIPTION
**Pull request by feifeiji89**
Wrap attribution scope fields into a dictionary for clarity.

***
<a href="https://pr-preview.s3.amazonaws.com/feifeiji89/attribution-reporting-api/pull/1391.html" title="Last updated on Aug 12, 2024, 8:28 PM UTC (6611cba)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1391/aad5718...feifeiji89:6611cba.html" title="Last updated on Aug 12, 2024, 8:28 PM UTC (6611cba)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -2651,30 +2662,44 @@ non-negative integer |defaultBudget|, and an [=aggregatable debug reporting conf
     with |value|, |budget|, |supportedTypes|, and |defaultConfig|.
 1. Return the [=tuple=] (|budget|, |config|).
 
-To <dfn>parse max event states</dfn> from a [=map=] |map| and
-a possibly null 32-bit positive integer |attributionScopeLimit|:
+To <dfn>parse attribution scope data</dfn> from a [=map=] |map|:
+1. If |map|["<code>[=source-registration JSON key/attribution_scope_data=]</code>"] does not [=map/exists|exist=], return null.
```
Comment (apasel422) [inline index.bs:2666]: What do you think about the following naming scheme, to avoid redundancy:

```jsonc
"attribution_scopes": {
  "values": [...], // was top-level attribution_scopes
  "limit": ...,    // was top-level attribution_scope_limit
  "max_event_states": ...
}
```

File: index.bs [hunk 2]
```diff
@@ -1384,14 +1394,14 @@ with a given ([=aggregatable debug rate-limit record/context site=], [=aggregata
 per [=aggregatable debug rate-limit window=]. Its value is (2<sup>20</sup>, 65536).
 
 <dfn>Default max event states</dfn> is a positive integer that controls the
```
Comment (apasel422) [inline index.bs:1396]: Is the concept of default max event states even necessary anymore? We might be better off requiring the `max_event_states` field to be given an explicit value rather than assume that the surrounding registration uses the default event configuration.

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: index.bs [hunk 3]
```diff
@@ -838,12 +852,8 @@ An attribution source is a [=struct=] with the following items:
...
 : <dfn>destination limit priority</dfn>
 :: A 64-bit integer.
-: <dfn>attribution scope limit</dfn>
-:: Null or a positive 32-bit integer representing the number of distinct [=attribution source/attribution scopes=] allowed per [=attribution source/attribution destinations|attribution destination=] for the source's [=attribution source/reporting origin=].
-: <dfn>attribution scopes</dfn>
-:: A [=set=] of [=strings=].
-: <dfn>max event states</dfn>
-:: A positive integer representing the maximum number of [=trigger states=] for [=source type/event=] sources per [=attribution source/attribution destinations|attribution destination=] for the source's [=attribution source/reporting origin=].
+: <dfn>attribution scope data</dfn> (default null)
+:: An [=attribution scope data=].
```
Comment (linnan-github) [inline index.bs:856]:
```suggestion
:: Null or an [=attribution scope data=].
```

File: index.bs (hunk 1 above)
Comment (linnan-github) [reply index.bs:2666]: +1

File: index.bs [hunk 4]
```diff
@@ -2750,15 +2775,7 @@ To <dfn noexport>parse source-registration JSON</dfn> given a [=byte sequence=]
...
-1. Let |attributionScopeLimit| be null.
-1. If |value|["<code>[=source-registration JSON key/attribution_scope_limit=]</code>"] [=map/exists=]:
-    1. Set |attributionScopeLimit| to |value|["<code>[=source-registration JSON key/attribution_scope_limit=]</code>"].
-    1. If |attributionScopeLimit| is not an integer, cannot be represented by an unsigned 32-bit integer, or is less than or equal to zero, return null.
-1. Let |maxEventStates| be the result of running
-    [=parse max event states=] with |value| and |attributionScopeLimit|.
-1. If |maxEventStates| is an error, return null.
-1. Let |attributionScopes| be the result of running [=parse attribution scopes for source=] with |value| and |attributionScopeLimit|.
-1. If |attributionScopes| is an error, return null.
+1. Let |attributionScopeData| be the result of running [=parse attribution scope data=] with |value|.
```
Comment (linnan-github) [inline index.bs:2778]: Null should be returned if `|attributionScopeData|` is an error.

**Comment by linnan-github** [comment]
Please also update explainer for the API surface change.

File: index.bs (hunk 1 above)
Comment (feifeiji89) [reply index.bs]: Done. this indeed is much simpler. thanks!

File: index.bs (hunk 2 above)
Comment (feifeiji89) [reply index.bs:1396]: Can we address this in a separate PR? was hoping to make this PR a pure refactor so that we can check it in sooner and we can move forward with the other 2 PRs (info gain, source processing).

Comment (feifeiji89) [reply index.bs:1396]: Synced offline. Let's keep this as-is for now, filed b/359253451 to track this.

File: attribution_scopes.md [hunk 5]
```diff
@@ -31,38 +31,40 @@ In general, the approach here is to allow API callers to specify a list of strin
 
 ### API changes
 
-The following optional parameters will be added to the JSON in `Attribution-Reporting-Register-Source` during source registration: `attribution_scopes`, `attribution_scope_limit`, and `max_event_states`:
+The following optional parameters will be added to the JSON in `Attribution-Reporting-Register-Source` during source registration: `attribution_scopes`, `values`, `limit`, and `max_event_states`:
```
Comment (apasel422) [inline attribution_scopes.md:34]: I would phrase this a bit differently, as `values` and `limit` are actually required within an `attribution_scopes` dictionary.

File: attribution_scopes.md [hunk 6]
```diff
@@ -31,38 +31,40 @@ In general, the approach here is to allow API callers to specify a list of strin
...
-  // Defaults to 3 if omitted.
-  // The flexible event-level script linked in the Privacy Considerations section below can be used to
-  // calculate number of states based on a configuration.
-  "max_event_states": <int>
+  "attribution_scopes" {
+    // Required
+    // Represents the total number of distinct scopes allowed per destination for the source reporting origin.
+    // This is used to calculate the information gain for event-level reports.
+    // Attribution scope limit values must be positive integers.  
+    // Defaults to null if omitted.
```
Comment (apasel422) [inline attribution_scopes.md:46]: Isn't this field required now?

File: attribution_scopes.md [hunk 7]
```diff
@@ -82,21 +84,21 @@ The following optional parameter will be added to the JSON in  `Attribution-Repo
...
+If there are multiple sources whose `attribution_scopes/values` contains at least one of the trigger’s `attribution_scopes`, then attribution will take place among the sources within the particular `attribution_scopes` according to current [Attribution Reporting API logic](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#trigger-attribution-algorithm).
 
 If the trigger registration's `attribution_scopes` is empty, then all sources are considered for attribution.
 
-Once an `attribution_scope_limit` is set, the last K values (where K = `attribution_scope_limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
+Once an `limit` is set, the last K values (where K = `limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
```
Comment (apasel422) [inline attribution_scopes.md:93]:
```suggestion
Once a `limit` is set, the last K values (where K = `limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
```

File: attribution_scopes.md [hunk 8]
```diff
@@ -31,38 +31,40 @@ In general, the approach here is to allow API callers to specify a list of strin
...
-  // subsequent event-source registrations.
-  // Example: default event source for event-level reports supports 1 attribution report, 1 reporting window,
-  // and 1 bit of trigger data for a total of 3 event states.
-  // This is used to calculate the information gain for event-level reports.
-  // Max event states must be positive integers.
-  // Defaults to 3 if omitted.
-  // The flexible event-level script linked in the Privacy Considerations section below can be used to
-  // calculate number of states based on a configuration.
-  "max_event_states": <int>
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:41]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 9]
```diff
@@ -106,44 +108,52 @@ If the current trigger passes the top-level filter check during the attribution
...
+This example shows an API caller that manages 3 advertisers that all sell products on the same destination site (scheme + eTLD+1). In this example the API caller uses an `limit` of 3 and tracks each advertiser with a distinct `attribution_scopes` value. Additionally, the user browsing the web sees 4 different ads from these advertisers and each one has a source registration associated:
 
 ```jsonc
 // source registration 1 for advertiser1 at t=0
 {
   ..., // existing fields
-  "attribution_scope_limit": 3,
-  "attribution_scopes": ["advertiser1"],
-  "max_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:117]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 10]
```diff
@@ -106,44 +108,52 @@ If the current trigger passes the top-level filter check during the attribution
...
 ```
 
 ```jsonc
 // source registration 2 for advertiser1 at t=1
 {
   ..., // existing fields
-  "attribution_scope_limit": 3,
-  "attribution_scopes": ["advertiser1"],
-  "max_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:129]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 11]
```diff
@@ -106,44 +108,52 @@ If the current trigger passes the top-level filter check during the attribution
...
 ```
 
 ```jsonc
 // source registration 3 for advertiser2 at t=2
 {
   ..., // existing fields
-  "attribution_scope_limit": 3,
-  "attribution_scopes": ["advertiser2"],
-  "max_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:141]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 12]
```diff
@@ -106,44 +108,52 @@ If the current trigger passes the top-level filter check during the attribution
...
 }
 ```
 ```jsonc
 // source registration 4 for advertiser3 at t=3
 {
   ..., // existing fields
-  "attribution_scope_limit": 3,
-  "attribution_scopes": ["advertiser3"],
-  "max_num_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:152]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 13]
```diff
@@ -153,11 +163,13 @@ The user then converts at a later time on the destination site by purchasing a p
 // trigger registration 1 for advertiser1 at t=4
 {
   ..., // existing fields
-  "attribution_scopes": ["advertiser1"]
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:166]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 14]
```diff
@@ -167,29 +179,35 @@ This example shows an API caller that manages an ad banner that contains multipl
 // source registration 1 for campaign promoting products 1, 2, and 3 at t=0
 {
   ..., // existing fields
-  "attribution_scope_limit": 4,
-  "attribution_scopes": ["product1", "product2", "product3"],
-  "max_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:182]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 15]
```diff
@@ -167,29 +179,35 @@ This example shows an API caller that manages an ad banner that contains multipl
...
 ```
 
 ```jsonc
 // source registration 2 for campaign promoting only product 2 at t=1
 {
   ..., // existing fields
-  "attribution_scope_limit": 4,
-  "attribution_scopes": ["product2"],
-  "max_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:194]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 16]
```diff
@@ -167,29 +179,35 @@ This example shows an API caller that manages an ad banner that contains multipl
...
 ```
 
 ```jsonc
 // source registration 3 for campaign promoting only product 4 at t=2
 {
   ..., // existing fields
-  "attribution_scope_limit": 4,
-  "attribution_scopes": ["product4"],
-  "max_event_states": 3
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:206]:
```suggestion
  "attribution_scopes": {
```

File: attribution_scopes.md [hunk 17]
```diff
@@ -199,11 +217,13 @@ The user then converts at a later time on the destination site and makes a purch
 // trigger registration 1 for product related to products 2 and 3 at t=3
 {
   ..., // existing fields
-  "attribution_scopes": ["product2", "product3"]
+  "attribution_scopes" {
```
Comment (apasel422) [inline attribution_scopes.md:220]:
```suggestion
  "attribution_scopes": {
```

File: ts/src/header-validator/to-json.ts [hunk 18]
```diff
@@ -184,9 +200,7 @@ type Source = CommonDebug &
     source_event_id: string
     trigger_data_matching: string
     aggregatable_debug_reporting?: SourceAggregatableDebugReportingConfig
-    attribution_scopes?: string[]
-    attribution_scope_limit?: number
-    max_event_states?: number
+    attribution_scope_data?: AttributionScopes
```
Comment (apasel422) [inline ts/src/header-validator/to-json.ts:203]:
```suggestion
    attribution_scopes?: AttributionScopes
```

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: ts/src/header-validator/source.test.ts [hunk 19]
```diff
@@ -2687,150 +2687,158 @@ const testCases: TestCase[] = [
...
-        path: ['max_event_states'],
-        msg: 'must be default (3) if attribution_scope_limit is not set',
+        path: ['attribution_scopes', 'values'],
+        msg: 'must be non-empty if limit is set',
       },
     ],
   },
   {
-    name: 'invalid-attribution-scope-limit-attribution-scopes',
+    name: 'missing-attribution-scope-limit',
```
Comment (apasel422) [inline ts/src/header-validator/source.test.ts:2788]: 1. Please also add a test case for a missing `values` field.
2. Please also add a test case for the `attribution_scopes` field not being a dictionary.

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: ts/src/header-validator/source.test.ts (hunk 19 above)
Comment (feifeiji89) [reply ts/src/header-validator/source.test.ts:2802]: Nice catch. Done.

**Review by apasel422 — APPROVED** [review]

File: attribution_scopes.md [hunk 20]
```diff
@@ -82,21 +83,21 @@ The following optional parameter will be added to the JSON in  `Attribution-Repo
...
+If there are multiple sources whose `attribution_scopes/values` contains at least one of the trigger’s `attribution_scopes`, then attribution will take place among the sources within the particular `attribution_scopes` according to current [Attribution Reporting API logic](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#trigger-attribution-algorithm).
 
 If the trigger registration's `attribution_scopes` is empty, then all sources are considered for attribution.
 
-Once an `attribution_scope_limit` is set, the last K values (where K = `attribution_scope_limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
+Once a `limit` is set, the last K values (where K = `limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
```
Comment (linnan-github) [inline attribution_scopes.md:92]: Should this refer to values as well?

Comment (linnan-github) [inline attribution_scopes.md:92]: Is there a reason we use attribution_scopes/values but just limit instead of attribution scopes/limit?

File: attribution_scopes.md [hunk 21]
```diff
@@ -106,44 +107,52 @@ If the current trigger passes the top-level filter check during the attribution
 
 ### Example 1: distinct attribution scopes
 
-This example shows an API caller that manages 3 advertisers that all sell products on the same destination site (scheme + eTLD+1). In this example the API caller uses an `attribution_scope_limit` of 3 and tracks each advertiser with a distinct `attribution_scopes` value. Additionally, the user browsing the web sees 4 different ads from these advertisers and each one has a source registration associated:
+This example shows an API caller that manages 3 advertisers that all sell products on the same destination site (scheme + eTLD+1). In this example the API caller uses an `limit` of 3 and tracks each advertiser with a distinct `attribution_scopes` value. Additionally, the user browsing the web sees 4 different ads from these advertisers and each one has a source registration associated:
```
Comment (linnan-github) [inline attribution_scopes.md:110]: Should this refer to values instead of attribution_scopes?

File: attribution_scopes.md [hunk 22]
```diff
@@ -153,11 +162,13 @@ The user then converts at a later time on the destination site by purchasing a p
 // trigger registration 1 for advertiser1 at t=4
 {
   ..., // existing fields
-  "attribution_scopes": ["advertiser1"]
+  "attribution_scopes": {
```
Comment (linnan-github) [inline attribution_scopes.md:165]: I think we only change the source side?

**Review by linnan-github — CHANGES_REQUESTED** [review]

File: index.bs [hunk 23]
```diff
@@ -2651,35 +2662,49 @@ non-negative integer |defaultBudget|, and an [=aggregatable debug reporting conf
...
+1. Let |attributionScopes| be a new [=attribution scopes=] with the items:
+    : [=attribution scopes/limit=]
+    :: |limit|
+    : [=attribution scopes/values=]
+    :: |values|
+    : [=attribution scopes/max event states=]
+    :: |maxEventStates|
+1. Return |attributionScopes|.
+
+To <dfn>parse max event states</dfn> from a [=map=] |map|:
```
Comment (linnan-github) [inline index.bs:2685]: We can simply return default value if it doesn’t exist.

File: index.bs [hunk 24]
```diff
@@ -2651,35 +2662,49 @@ non-negative integer |defaultBudget|, and an [=aggregatable debug reporting conf
...
-1. If |maxEventStates| is not equal to [=default max event states=] and |attributionScopeLimit| is null, return an error.
 1. Return |maxEventStates|.
 
-To <dfn>parse attribution scopes for source</dfn> from a [=map=] |map| and a possibly null 32-bit positive integer |attributionScopeLimit|:
+To <dfn>parse attribution scope values for source</dfn> from a [=map=] |map| and a 32-bit positive integer |limit|:
+1. If |map|["<code>[=source-registration JSON key/values=]</code>"] does not [=map/exist=], return an error.
 1. Let |result| be a new [=set=].
-1. If |map|["<code>[=source-registration JSON key/attribution_scopes=]</code>"] does not [=map/exist=], return |result|.
-1. Let |values| be |map|["<code>[=source-registration JSON key/attribution_scopes=]</code>"].
+1. Let |values| be |map|["<code>[=source-registration JSON key/values=]</code>"].
```
Comment (linnan-github) [inline index.bs:2695]: We can return null earlier if the list is empty.

File: attribution_scopes.md (hunk 20 above)
Comment (feifeiji89) [reply attribution_scopes.md]: changed this to attribution_scopes/xxx.

File: attribution_scopes.md [hunk 25]
```diff
@@ -31,38 +31,39 @@ In general, the approach here is to allow API callers to specify a list of strin
 
 ### API changes
 
-The following optional parameters will be added to the JSON in `Attribution-Reporting-Register-Source` during source registration: `attribution_scopes`, `attribution_scope_limit`, and `max_event_states`:
+The following optional parameter will be added to the JSON in `Attribution-Reporting-Register-Source` during source registration: `attribution_scopes`, which contains required parameters `values`, `limit`, and optional parameter `max_event_states`:
```
Comment (linnan-github) [inline attribution_scopes.md:34]:
```suggestion
The following optional parameter will be added to the JSON in `Attribution-Reporting-Register-Source` during source registration: `attribution_scopes`, which contains required parameters `values`, `limit`, and an optional parameter `max_event_states`:
```

File: attribution_scopes.md [hunk 26]
```diff
@@ -82,21 +83,21 @@ The following optional parameter will be added to the JSON in  `Attribution-Repo
...
+If there are multiple sources whose `attribution_scopes/values` contains at least one of the trigger’s `attribution_scopes`, then attribution will take place among the sources within the particular `attribution_scopes` according to current [Attribution Reporting API logic](https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#trigger-attribution-algorithm).
 
 If the trigger registration's `attribution_scopes` is empty, then all sources are considered for attribution.
 
-Once an `attribution_scope_limit` is set, the last K values (where K = `attribution_scope_limit`) of `attribution_scopes` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes` values will be treated as if the attribution scopes were empty.
+Once a `attribution scopes/limit` is set, the last K values (where K = `attribution scopes/limit`) of `attribution_scopes/values` will be considered the final set of `attribution_scopes` values and any source with additional `attribution_scopes/values` will be treated as if the attribution scopes were empty.
```
Comment (linnan-github) [inline attribution_scopes.md:92]:
```suggestion
Once a `attribution scopes/limit` is set, the last K values (where K = `attribution scopes/limit`) of `attribution_scopes/values` will be considered the final set of attribution scopes values and any source with additional values will be treated as if the attribution scopes were not set.
```

File: attribution_scopes.md [hunk 27]
```diff
@@ -153,11 +162,11 @@ The user then converts at a later time on the destination site by purchasing a p
 // trigger registration 1 for advertiser1 at t=4
 {
   ..., // existing fields
-  "attribution_scopes": ["advertiser1"]
+   "attribution_scopes": ["advertiser1"]
```
Comment (linnan-github) [inline attribution_scopes.md:165]:
```suggestion
  "attribution_scopes": ["advertiser1"]
```

File: attribution_scopes.md [hunk 28]
```diff
@@ -153,11 +162,11 @@ The user then converts at a later time on the destination site by purchasing a p
...
 }
 ```
 
-The API automatically performs attribution between any sources that have `attribution_scopes` that are not disjoint with the trigger `attribution_scopes`. Any sources that do not have an `attribution_scopes` that matches at least one of the trigger registration `attribution_scopes` are deleted (assuming the source that is chosen passes the top-level filter check; if it does not then no sources are deleted). In this example, the API caller would receive an attribution report attributing the trigger registration to advertiser1’s second source registration.
+The API automatically performs attribution between any sources that have `attribution_scopes/values` that are not disjoint with the trigger `attribution_scopes`. Any sources that do not have an `attribution_scopes` that matches at least one of the trigger registration `attribution_scopes` are deleted (assuming the source that is chosen passes the top-level filter check; if it does not then no sources are deleted). In this example, the API caller would receive an attribution report attributing the trigger registration to advertiser1’s second source registration.
```
Comment (linnan-github) [inline attribution_scopes.md:169]:
```suggestion
The API automatically performs attribution between any sources that have `attribution_scopes/values` that are not disjoint with the trigger `attribution_scopes`. Any sources that do not have an `attribution_scopes/values` that matches at least one of the trigger registration `attribution_scopes` are deleted (assuming the source that is chosen passes the top-level filter check; if it does not then no sources are deleted). In this example, the API caller would receive an attribution report attributing the trigger registration to advertiser1’s second source registration.
```

File: attribution_scopes.md [hunk 29]
```diff
@@ -199,11 +214,13 @@ The user then converts at a later time on the destination site and makes a purch
 // trigger registration 1 for product related to products 2 and 3 at t=3
 {
   ..., // existing fields
-  "attribution_scopes": ["product2", "product3"]
+  "attribution_scopes": {
```
Comment (linnan-github) [inline attribution_scopes.md:217]: Please revert the change for triggers.

File: index.bs [hunk 30]
```diff
@@ -2651,35 +2662,49 @@ non-negative integer |defaultBudget|, and an [=aggregatable debug reporting conf
...
 1. If |values| is not a [=list=], return an error.
+1. If |values| [=set/is empty=], return an error.
```
Comment (linnan-github) [inline index.bs:2697]:
```suggestion
1. If |values| [=list/is empty=], return an error.
```

**Review by linnan-github — APPROVED** [review]

File: attribution_scopes.md [hunk 31]
```diff
@@ -218,14 +233,14 @@ The use of `attribution_scopes` will have some impact on the amount of utility t
...
 
-TODO: Update script to allow for testing different configurations that include attribution_scope_limit
+TODO: Update script to allow for testing different configurations that include `attribution scopes/limit`
 
-Note that the `attribution_scope_limit` does not have any impact on the privacy mechanisms used in Aggregatable Reports.
+Note that the `attribution scopes/limit` does not have any impact on the privacy mechanisms used in Aggregatable Reports.
 
 ### Additional Privacy Limits
-
+s
```
Comment (apasel422) [inline attribution_scopes.md:243]:
```suggestion
```
